lib_collections/utils.py

```diff
@@ -1,377 +1,711 @@
 """
 iiifcollectionbrowse
 """
-from threading import Thread
-from urllib.parse import unquote
 
-import requests
-
-from pyiiif.pres_api.utils import get_thumbnail
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
 
-from .exceptions import (
-    IncompatibleRecordError, InvalidCollectionRecordError,
-    NoCollectionFoundError, NoCollectionParameterError
-)
+from django.utils.text import slugify
+from django.http.response import Http404
+from urllib.parse import urlencode
 
-config = {
-    "VIEWER_URL": "https://iiif-viewer.lib.uchicago.edu/uv/uv.html#",
-    "REQUESTS_TIMEOUT": 1,
-    "NO_THUMB_IMG_URL": '',
-    "CONTRAST_COLOR": "#800000",
-    "THUMBNAIL_BACKDROP": "#D6D6CE"
-}
+import requests
+import re
+import json
+import os
 
-VIEWER_URL = config['VIEWER_URL']
-REQUESTS_TIMEOUT = float(config['REQUESTS_TIMEOUT'])
-NO_THUMB_IMG_URL = config['NO_THUMB_IMG_URL']
-THUMBS_PER_PAGE = 10
-COLORS = {
-    "contrast_color": config['CONTRAST_COLOR'],
-    "thumbnail_backdrop": config['THUMBNAIL_BACKDROP']
-}
+from citeproc import CitationStylesStyle, CitationStylesBibliography
+from citeproc import Citation, CitationItem
+from citeproc import formatter
+from citeproc.source.json import CiteProcJSON
 
+LANGUAGE_ABBREVS = {'en': 'English'}
 
-def threaded_thumbnails(identifier, result, index):
-    """
-    Wraps get_thumbnail() for multi-threaded solutions
+# toggle this comment to go between dev and production
+IIIF_PREFIX = "https://iiif-collection.lib.uchicago.edu"
+# IIIF_PREFIX = "https://iiif-collection-dev.lib.uchicago.edu"
 
-    The array should be "initialized" to the correct size,
-    and each thread should deposit it's thumbnail into the
-    appropriate index.
+WAGTAIL_PREFIX = "/collex/collections"
 
-    Args:
-        identifier: string
+# toggle these root URLs for dev/production
+# CITATION_ROOT = "http://localhost:9000"
+CITATION_ROOT = "http://www.lib.uchicago.edu/cgi-bin/citation"
 
-        result: list
+TURTLE_ROOT = "https://ark.lib.uchicago.edu/ark:61001/"
 
-        index: int
 
-    Returns:
-        None
+class CBrowseURL():
```

Comment:
Good idea--thanks.  Yeah, I was sort of experimenting here with using classes exclusively for namespacing.  (Is that what Tim Peters had in mind with the last line of the Zen of Python?  I still have no idea.)  It's a little bit of an awkward fit for the notion of an OOP class since there are no instances or inheritance at play, but on the other hand, Python doesn't let you define modules recursively (as far as I know).  So I guess I'll continue with the experiment for now...